### PR TITLE
feat: add support for `genesis download`

### PR DESCRIPTION
**Pull request by zivkovicmilos**
## Description

This PR introduces a `genesis download` subcommand, for downloading official `genesis.json` artifacts using the `gnoland` binary.

<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [x] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [x] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [x] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: gno.land/cmd/gnoland/genesis_download.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+	errChainNotSupported = errors.New("chain ID not supported")
+)
+
+const test4ID = "test4"
+
+const deploymentPathFormat = "https://raw.githubusercontent.com/gnolang/gno/refs/heads/master/misc/deployments/%s.gno.land/genesis.json"
+
+var genesisSHAMap = map[string]string{
+	test4ID: "beb781dffc09b96e3114fb7439fa85c4fe8ea796f64ec0cd3801a6b518ab023c",
+}
```
Comment (thehowl) [inline gno.land/cmd/gnoland/genesis_download.go:30]: This should be automated; or at least we should test the verification (ie. each source should map the sha sum, this should be an explicit test)

Also the link to genesis.json shouldn't be master, but a permalink like a commit. I think we can avoid making this dynamic with the `%s`, but instead have a struct with `Source string; Checksum string`

File: gno.land/cmd/gnoland/genesis_download.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+		return fmt.Errorf("unable to make GET request, %w", err)
+	}
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("unexpected status code %d", resp.StatusCode)
+	}
+
+	defer resp.Body.Close()
+
+	f, err := os.Create(
```
Comment (thehowl) [inline gno.land/cmd/gnoland/genesis_download.go:123]: os.CreateTemp?

**Comment by albttx** [comment]
I don't understand why you're downloading the Genesis file from github, the genesis file is downloadable from any RPCs, 

Btw if you check a [cosmoshub rpc](https://cosmos-rpc.polkachu.com/) they have an extra endpoint `/genesis_chunked?chunk=_`, genesis file could be pretty huge sometime, (some chains hardfork, and get all previous state as txs in genesis)

https://rpc.test4.gno.land/genesis

**Comment by moul** [comment]
Please wait for comments on this issue: https://github.com/gnolang/gno/issues/2824

I believe we should clean up gnoland by removing the genesis subcommand and moving the commands to a contribs binary.

**Comment by zivkovicmilos** [comment]
> I don't understand why you're downloading the Genesis file from github, the genesis file is downloadable from any RPCs,

A live RPC is not really the most reliable backup and restore mechanism, [as 100% of our deployments are proof of that](https://status.gnoteam.com/)

File: gno.land/cmd/gnoland/genesis_download.go (hunk 1 above)
Comment (zivkovicmilos) [reply gno.land/cmd/gnoland/genesis_download.go]: I adopted the metadata structure you were suggesting:

[e3fb205](https://github.com/gnolang/gno/pull/2841/commits/e3fb2051fa2ae12ddc11a382ac7857acaf87a178)

Our unit tests already cover actually fetching the genesis from the source and verifying it

File: gno.land/cmd/gnoland/genesis_download.go (hunk 2 above)
Comment (zivkovicmilos) [reply gno.land/cmd/gnoland/genesis_download.go:128]: Why would we use it instead of `os.Create`?

**Comment by zivkovicmilos** [comment]
> Please wait for comments on this issue: #2824
> 
> I believe we should clean up gnoland by removing the genesis subcommand and moving the commands to a contribs binary.

@moul 

We can always move this feature along with the command suite, I don't think it adds additional bloat we should worry about

**Comment by mazzy89** [comment]
I do not see much benefit of having this command around. In our deployment model the genesis is passed as a json file inside a k8s `ConfigMap` and mounted in a volume of the container pod. Can't imagine any very useful place to have this command in our current flow.

**Comment by mazzy89** [comment]
Actually looking better at our deployment flow, we pull the genesis at bootstrap time in an init container using curl. A simply `curl` command to pull it from Github or S3. I mean this command might be useful but nothing that a `curl` command does not already do.

**Comment by r3v4s** [comment]
@zivkovicmilos 
> I don't think it adds additional bloat we should worry about.

Agreed, this pr isn't really spaghetti

> what do you think about this sugar command?

I really don't have specific opinion for this, as @mazzy89 said we can achieve same result with `curl/wget`. 

However maybe we can use this command as internally for bootstrap script from our side. Or it can be used inside some docker image that doesn't support `curl/wget` by default.

**Comment by sw360cab** [comment]
> Actually looking better at our deployment flow, we pull the genesis at bootstrap time in an init container using curl. A simply `curl` command to pull it from Github or S3. I mean this command might be useful but nothing that a `curl` command does not already do.

IMO it can be useful if the goal is concentrating all the useful setup commands within the `gnoland` image.
From a DevOps POV this means referring always to this image, without leveraging other handy images, or requiring other extra tools within the image itself.

On the other hand, probably this subsubcommand is satisfying only a very small subset of use cases, the price of maintaining it can become higher that its real usefulness.

**Comment by zivkovicmilos** [comment]
Closing this PR after comments from the ops crew 🙏 

It's not a feature that should suck out maintain-time in the future :)

**Pull request closed without merging**